cranelift/filetests/filetests/runtests/br.clif

```diff
@@ -19,20 +68,161 @@ block2:
     v2 = bconst.b1 false
     return v2
 }
-; run
+; run: %brz_i16(0) == true
+; run: %brz_i16(1) == false
+; run: %brz_i16(-1) == false
 
-function u0:1() -> b1 {
+function %brz_i8(i8) -> b1 {
+block0(v0: i8):
+    brz v0, block1
+    jump block2
+
+block1:
+    v1 = bconst.b1 true
+    return v1
+
+block2:
+    v2 = bconst.b1 false
+    return v2
+}
+; run: %brz_i8(0) == true
+; run: %brz_i8(1) == false
+; run: %brz_i8(-1) == false
+
+
+; TODO: Merge this with brz_b1_false when we are able to pass bool imm's in test params
+function %brz_b1_true() -> b1 {
 block0:
-    v0 = iconst.i8 0
-    brnz v0, block1
+    v1 = bconst.b1 true
+    brz v1, block1
     jump block2
 
 block1:
+    v2 = bconst.b1 true
+    return v2
+
+block2:
+    v3 = bconst.b1 false
+    return v3
+}
+; run: %brz_b1_true() == false
+
+function %brz_b1_false() -> b1 {
+block0:
     v1 = bconst.b1 false
+    brz v1, block1
+    jump block2
+
+block1:
+    v2 = bconst.b1 true
+    return v2
+
+block2:
+    v3 = bconst.b1 false
+    return v3
+}
+; run: %brz_b1_false() == true
+
+
+
+function %brnz_i64(i64) -> b1 {
+block0(v0: i64):
+    brnz v0, block1
+    jump block2
+
+block1:
+    v1 = bconst.b1 true
     return v1
 
 block2:
+    v2 = bconst.b1 false
+    return v2
+}
+; run: %brnz_i64(0) == false
+; run: %brnz_i64(1) == true
+; run: %brnz_i64(-1) == true
+
+function %brnz_i32(i32) -> b1 {
+block0(v0: i32):
+    brnz v0, block1
+    jump block2
+
+block1:
+    v1 = bconst.b1 true
+    return v1
+
+block2:
+    v2 = bconst.b1 false
+    return v2
+}
+; run: %brnz_i32(0) == false
+; run: %brnz_i32(1) == true
+; run: %brnz_i32(-1) == true
+
+function %brnz_i16(i16) -> b1 {
+block0(v0: i16):
+    brnz v0, block1
+    jump block2
+
+block1:
+    v1 = bconst.b1 true
+    return v1
+
+block2:
+    v2 = bconst.b1 false
+    return v2
+}
+; run: %brnz_i16(0) == false
+; run: %brnz_i16(1) == true
+; run: %brnz_i16(-1) == true
+
+function %brnz_i8(i8) -> b1 {
+block0(v0: i8):
+    brnz v0, block1
+    jump block2
+
+block1:
+    v1 = bconst.b1 true
+    return v1
+
+block2:
+    v2 = bconst.b1 false
+    return v2
+}
+; run: %brnz_i8(0) == false
+; run: %brnz_i8(1) == true
+; run: %brnz_i8(-1) == true
+
+
+; TODO: Merge this with brz_b1_false when we are able to pass bool imm's in test params
```

Comment:
For reference, the issue that fixes this is #2237.

Reply:
(But also looking at that issue, I must have thought when I wrote it that we could at least pass `b1` and `b8`? Is that not the case anymore?)

Reply:
Tested this again now, curiously, it works if we enable just the `interpret` test, but stops working with the `run` test.

```
test interpret
test run
target x86_64 machinst

function %brnz_b1(b1) -> b1 {
block0(v1: b1):
    brnz v1, block1
    jump block2

block1:
    v2 = bconst.b1 true
    return v2

block2:
    v3 = bconst.b1 false
    return v3
}
; run: %brnz_b1(true) == true
; run: %brnz_b1(false) == false
```

We get:

```
     Running `C:\Users\Afonso\CLionProjects\wasmtime\target\debug\clif-util.exe test ./test.clif`
thread 'worker #0' panicked at 'argument type mismatch: b8 != b1', cranelift\filetests\src\function_runner.rs:200:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FAIL ./test.clif: panicked in worker #0: argument type mismatch: b8 != b1
1 tests
Error: 1 failure
```

Reply:
Well, I guess we'll just have to fix it all in #2237 😁